src/platform/Zephyr/DiagnosticDataProviderImpl.cpp

```diff
@@ -212,7 +212,30 @@ CHIP_ERROR DiagnosticDataProviderImpl::GetNetworkInterfaces(NetworkInterface **
             ifp->hardwareAddress = ByteSpan(ifp->MacAddress, addressSize);
         }
 
-        head = ifp;
+        // Assuming IPv6-only support
+        Inet::InterfaceAddressIterator interfaceAddressIterator;
+        chip::Inet::IPAddress ipv6Addresses[kMaxIPv6AddrCount];
+        uint8_t ipv6AddressesCount;
+        for (ipv6AddressesCount = 0; interfaceAddressIterator.HasCurrent(); ipv6AddressesCount++)
```

Comment:
Increment `ipv6AddressesCount` only when you add a new address. Otherwise, you could increment the count when `interfaceAddressIterator.GetInterfaceId() != interfaceIterator.GetInterfaceId()`